apps/examples/board_specific_demo/ameba_secure_world_demo/secure_world.c

```diff
@@ -0,0 +1,68 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+
+#define SHOW_USAGE printf("Usage: secure_world -r -w -a\n \
```

Comment:
The -a option allows us to generalize the example to work on different boards for their respective secure world regions.
So what is no address input is given, is there a default address that the test works for?

Also, can we add the Secure World regions of tp1e, etc to this usage help output so that the user can test conveniently?
Later, we can add the addresses of other boards too as and when needed so the user can print the help menu and then give the right input for testing.

Reply:
Take this into consideration please, it helps future proof the application.

We can make -a mandatory if we provide info to the user in the help menu about Secure world address ranges for testing.
The default values used currently would only work for Tp1e and perhaps not if the test was run on another board.

@sunghan-chang @abhishek-samsung  your opinion please?

Reply:
I don't think we need to support other boards as this is a board-specific example. We can add the appropriate default addresses based on configs like the secure storage example.

https://github.com/Samsung/TizenRT/blob/eca807f0c25b342b12b233f6fd6e3e63c0e7c4a9/apps/examples/board_specific_demo/ameba_secure_storage_demo/secure_storage.c#L46-L53

It will be easier for the user to execute the command without providing the address, so I think it's better to have a default address based on the board tp1x/tp1e.

> Also, can we add the Secure World regions of tp1e, etc to this usage help output so that the user can test conveniently?

I agree with this, we can add the region addresses in help according to the board tp1x/tp1e

Reply:
How about adding it in Kconfig because it's a different value for every board?
If we add it as Kconfig, we can keep the same code for every board. (No need to modify the code itself)